src/base/net/downloadmanager.cpp

```diff
@@ -104,11 +102,10 @@ class Net::DownloadManager::NetworkCookieJar final : public QNetworkCookieJar
     {
         const QDateTime now = QDateTime::currentDateTime();
         QList<QNetworkCookie> cookies = cookieList;
-        for (const QNetworkCookie &cookie : cookieList)
```

Comment:
This iteration seems problematic, maybe consider backport the fix to v4.6.0?

Reply:
> This iteration seems problematic, maybe consider backport the fix to v4.6.0?

👍 
Please extract this commit into its own PR.

Reply:
You can port the `Use efficient algorithm for removing entries` commit.

Reply:
1. Having a separate PR is better for keeping referring in backport PR to be uniform. 
2. Anyway it's better to split unrelated commits into different PRs.

Is it too hard to extract it?

Reply:
#19345